learning-bitmark/contributing-to-bitmark/bitmark-software-development-process.md

```diff
@@ -0,0 +1,127 @@
+
+![](https://github.com/bitmark-inc/docs/blob/master/learning-bitmark/contributing-to-bitmark/imgs/review-process.png)
+
+# Development work-flow improvements
+
+**Development process:**
+
+This is our working flow using GitHub. Here I describe 3 basic areas of our development process.
+
+
+![](https://github.com/bitmark-inc/docs/blob/master/learning-bitmark/contributing-to-bitmark/imgs/new-review-process.png)
+
+
+**What to do when there is a new patch?**
+
+First, we need to ask ourselves, is it a trivial patch? In case it is not a 3-lines change patch, it means, it is not a trivial patch, sometimes even a 3-lines change is not trivial. We do need to be mature enough to understand what is and what is not a trivial patch. In case it is not a trivial patch, we shall open a GitHub issue with a proper title and describe what is the problem we are trying to fix, following with a pull request with a patch (if we have one), this patch must point out to the issue number. All the discussion related to implementation, shall be in the issue, code problems shall be in the pull request.
+
+**I would like to stress a bit how to perform a proper code review:**
+
+We need to always remember, there are other people watching us, the information is on the Internet and it will always be there, we need developers to engage with us, perhaps we need companies out there to see us relevant and join our cause.
+
+**What to do:**
+
+1. Know what is under review, understand what the developer is trying to achieve.
+2. Know what to look for in code: Structure, Style, Logic, Performance, Test coverage, Design, Readability, Maintainability and Functionality.
+3. Does it pass on CircleCi/TravisCi and Jenkins? Otherwise don't need start to make a review.
+4. Build and test.
+5. Give feedback that helps and not hurt, be impartial, as much as somebody prefers BLUE, some other people prefer YELLOW, so be impartial.
+6. Communicate goals and expectations, we use Slack and if we have direct access to the person that opened the code review, instead to go and make a comment and wait the person reply, talk directly with the person, it saves time and foster a positive culture.
+7. Peer review is all about trust, we need to be glad somebody is asking us to review its code.
+8. Be nice, at the end of the day, there is no software made by only one person.
+9. If the pull request comes from an outsider contributor, be nicer than usual, even if it is necessary to fix any code before merge and give all the credits to him/her, we need this person to come back and start to contribute more often. So, don't be harsh with newcomers.
+10. Remember, when doing a code review it might takes 30 minutes, but the person that made the code might have spent weeks, so be empathetic.
+
+**What not to do:**
+
```

Comment:
And again here is another example where having standardized list style could really help. In particular, having all of these be "Don't"s can really underline the purpose.